### ISSUE TITLE
Error loading remotes with "_" in the entry name

### ISSUE THREAD

**Issue by RussellCanfield** [module-federation/vite#85]
Thanks for all the hard work on the plugin! I was testing a few scenarios and ran into an issue where I have a remote name "vite_webpack" and it was replacing the "_" with "/" in the remote name and causing some failures.

The code causing the issue is here:

https://github.com/module-federation/vite/blob/4291753aaddc6ee54a6439ad9525cb7721f1f0e5/src/plugins/pluginProxyRemotes.ts#L41

I am not sure all the cases to cover here, but I was thinking maybe we can solve it like this:

```typescript
matchRemotesList.forEach((matchValue) => {
          const parts = devRemoteModuleName?.split(matchValue);
          if (!parts) return;
          const processedParts = parts.map((part, index) => {
            if (index === 0) {
              // Process the part before matchValue
              return part.replace(/\//g, '').replace(/_/g, '/').replace('.js', '');
            } else {
              // Process the part after matchValue
              return matchValue + part.replace(/\//g, '').replace(/_/g, '/').replace('.js', '');
            }
          });
          const processedDevRemoteModuleName = processedParts.join('');
          if (!processedDevRemoteModuleName) return;
          return generateRemotes(processedDevRemoteModuleName, command);
        });
```

Basically only try to normalize/parse the pieces after the remove name or before. But not the actual remote name. I am not sure all the implications to call it a fix :)

**Comment by zhangHongEn** [comment]
Thank you for your fix, but I have [deleted this code](https://github.com/zhangHongEn/vite/blob/feature/mfmanifest/src/plugins/pluginProxyRemoteEntry.ts) recently when I was compatible with nuxt. I will release a new version in the next two days. The latest code will not have this problem.

**Comment by gioboa** [comment]
That's great @zhangHongEn 
Looking forward to seeing your PR

**Comment by zhangHongEn** [comment]
Sorry, I need to clarify that I haven't completed nuxt yet. The next PR will support manifestion.json

---- Replied Message ----
| From | Giorgio ***@***.***> |
| Date | 09/15/2024 00:43 |
| To | module-federation/vite ***@***.***> |
| Cc | zhn ***@***.***>,
Mention ***@***.***> |
| Subject | Re: [module-federation/vite] Error loading remotes with "_" in the entry name (Issue #85) |

That's great @zhangHongEn
Looking forward to seeing your PR

**Comment by gioboa** [comment]
Closed by #86
